drencher/src/solver/mod_count.rs

```diff
@@ -0,0 +1,23 @@
+use super::{Solver, Solution};
+use board::Board;
+use color::Color;
+
+pub struct ModCount;
+
+impl Solver for ModCount {
+    fn solve(&self, mut b: Board) -> Result<Solution, Solution> {
+        let mut solution = Solution::new();
+        let mut index = 0u8;
+
+        while !b.is_drenched() {
+            let color = Color::new(index);
+            solution.push(color);
+            b.drench(color);
+
+            index += 1;
+            index %= 6;
+        }
+
+        Ok(solution)
+    }
```

Comment:
Let's make it more functional and rusty! How about:

```rust
Ok(
    (0..)
        .map(|i| Color::new(i % 6))
        .take_while(|_| !b.is_drenched())
        .inspect(|&c| b.drench(c))
        .collect()
)
```

Just wrote it, not entirely sure if it works...